jboss-eap-config-elytron/added/launch/elytron.sh

```diff
@@ -0,0 +1,261 @@
+}
+
+has_elytron_legacy_tls() {
+    declare config_file="$1"
+    if grep -q '<!-- ##TLS## -->' "${config_file}"
+    then
+        echo "true"
+    else
+        echo "false"
+    fi
+}
+
+has_elytron_keystore() {
+    declare config_file="$1"
+    if grep -q '<!-- ##ELYTRON_KEY_STORE## -->' "${config_file}"
+    then
+        echo "true"
+    else
+        echo "false"
+    fi
+}
+
+insert_elytron_tls() {
+ # the elytron skelton config. This will be used to replace <!-- ##ELYTRON_TLS## -->
+ # if this is replaced, we'll also remove the legacy <!-- ##TLS## --> marker
+ local elytron_tls="\
+        <tls>\n\
+            <key-stores>\n\
+                <!-- ##ELYTRON_KEY_STORE## -->\n\
+            </key-stores>\n\
+            <key-managers>\n\
+                <!-- ##ELYTRON_KEY_MANAGER## -->\n\
+            </key-managers>\n\
+            <server-ssl-contexts>\n\
+                <!-- ##ELYTRON_SERVER_SSL_CONTEXT## -->\n\
+            </server-ssl-contexts>\n\
+         </tls>\n"
+    # check for new config tag, use that if it's present
+    if [ "true" = $(has_elytron_tls "${CONFIG_FILE}") ]; then
+        sed -i "s|<!-- ##ELYTRON_TLS## -->|${elytron_tls}|" $CONFIG_FILE
+        # remove the legacy tag, if it's present
+        sed -i "s|<!-- ##TLS## -->||" $CONFIG_FILE
+    fi
+}
+
+elytron_legacy_config() {
+    declare https_keystore="$1" https_keystore_type="$2" https_password="$3" https_key_password="$4" https_keystore_dir="$5"
+
+    local keystore_path=""
+    local keystore_rel_to=""
+    local key_password=""
+
+    if [ -n "${https_key_password}" ]; then
+      key_password=${https_key_password}
+    else
+      key_password=${https_password}
+    fi
+
+    if [ -z "${https_keystore_dir}"  ]; then
+      # Documented behavior; HTTPS_KEYSTORE is relative to the config dir
+      # Use case is the user puts their keystore in their source's 'configuration' dir and s2i pulls it in
+      keystore_path="path=\"${https_keystore}\""
+      keystore_rel_to="relative-to=\"jboss.server.config.dir\""
+    elif [[ "${https_keystore_dir}" =~ ^/ ]]; then
+      # Assume leading '/' means the value is a FS path
+      # Standard template behavior where the template sets this var to /etc/eap-secret-volume
+      keystore_path="path=\"${https_keystore_dir}/${https_keystore}\""
+      keystore_rel_to=""
+    else
+      # Compatibility edge case. Treat no leading '/' as meaning HTTPS_KEYSTORE_DIR is the name of a config model path
+      keystore_path="path=\"${https_keystore}\""
+      keystore_rel_to="relative-to=\"${https_keystore_dir}\""
+    fi
+
+    local elytron_key_store=$(create_elytron_keystore "LocalhostKeyStore" "${https_keystore}" "${https_password}" "${https_keystore_type}" "${https_keystore_dir}")
+    local elytron_key_manager=$(create_elytron_keymanager "LocalhostKeyManager" "LocalhostKeyStore" "${key_password}")
+    local elytron_server_ssl_context=$(create_elytron_ssl_context "LocalhostSslContext" "LocalhostKeyManager")
```

Comment:
I think everything from L75-L101 can be removed and then L103-105 can be replaced with a 'declare' where the already calculated values from configure_https() are passed in.  The only use of this function I see is L226, and there we already have the elements calculated.